app/app/Http/Controllers/SignUpController.php

```diff
@@ -71,14 +77,25 @@ public function createUser(Request $request)
 
     public function confirmEmail($user_email, $email_verification_token)
     {
+        $after_signup_redirect = Input::get('after_signup_redirect');
```

Comment:
I like that you're getting the redirect value from input here but it looks like you're ignoring it everywhere else which could lead to bugs.

When signing up without the add-location redirect, does it correctly stay on the profile or does it incorrectly redirect to add-location?

Reply:
Just tested it and it works properly. The way signup is setup right now is that it sends you to a page which then tells you to sign in and not to your profile after you click the confirmation link. The button that tells you to sign in will send you to the profile unless the redirect is there

Reply:
Instead of hardcoding the /add-location in the redirect below, you should use $after_signup_redirect.  If anyone later passes a different URL segment to confirmEmail, it should redirect there instead of /add-location.

Something like this:
```
'redirect' => '/signin?after_signin_redirect=' . $after_signup_redirect
```